src/phases/show-ability-phase.ts

```diff
@@ -1,14 +1,20 @@
 import { globalScene } from "#app/global-scene";
 import type { BattlerIndex } from "#app/battle";
 import { PokemonPhase } from "./pokemon-phase";
+import { getPokemonNameWithAffix } from "#app/messages";
 
 export class ShowAbilityPhase extends PokemonPhase {
   private passive: boolean;
+  private pokemonName: string;
+  private abilityName: string;
 
   constructor(battlerIndex: BattlerIndex, passive: boolean = false) {
     super(battlerIndex);
 
     this.passive = passive;
+    // Set these now as the pokemon object may change before the queued phase is run
+    this.pokemonName = getPokemonNameWithAffix(this.getPokemon());
+    this.abilityName = passive ? this.getPokemon().getPassiveAbility().name : this.getPokemon().getAbility().name;
```

Comment:
```suggestion
    this.abilityName = (passive ? this.getPokemon().getPassiveAbility() : this.getPokemon().getAbility()).name;
```
Could do this, not sure if it's better. 🤔

Reply:
Little smoother, I like it